src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/editor.js

```diff
@@ -324,7 +345,16 @@ function VisualizeAppController(
   // appState then they won't be equal.
   if (!_.isEqual(stateContainer.getState().vis, stateDefaults.vis)) {
     try {
-      vis.setState(stateContainer.getState().vis);
+      const visState = stateContainer.getState().vis;
+      const state = {
+        type: visState.type,
+        title: visState.title,
+        params: visState.params,
+        data: {
+          aggs: visState.aggs,
+        },
+      };
+      vis.setState(state);
```

Comment:
Seems you skip this suggestion (at least didn't see an answer), but I think it will simplify code.
If you only need to extract aggs and put them into data, you could simplify this:

const { vis: { aggs, ...visState } }= stateContainer.getState();
vis.setState({ ...visState, data: { aggs } });